packages/client/src/base-client.ts

```diff
@@ -67,6 +69,18 @@ export class BaseClient {
     return await this.connection.withDeadline(deadline, fn);
   }
 
+  /**
+   * Set an {@link https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal | `AbortSignal`} that, when aborted,
+   * cancels any ongoing requests executed in `fn`'s scope.
```

Comment:
nit: in other docstrings in this file we refer to "service requests" -- could bring them all in line with each other.